pkg/odo/cli/url/create.go

```diff
@@ -204,52 +210,62 @@ func (o *URLCreateOptions) Complete(name string, cmd *cobra.Command, args []stri
 
 // Validate validates the URLCreateOptions based on completed values
 func (o *URLCreateOptions) Validate() (err error) {
-	// Check if exist
-	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(o.DevfilePath) {
+	if !util.CheckOutputFlag(o.OutputFlag) {
+		return fmt.Errorf("given output format %s is not supported", o.OutputFlag)
+	}
+
+	// if experimental mode is enabled, and devfile is provided.
+	errorList := make([]string, 0)
+	if o.isDevFile {
 		// check if a host is provided for route based URLs
-		if o.isRouteSupported && !o.wantIngress && len(o.host) > 0 {
-			return fmt.Errorf("host is not supported for URLs of Route Kind")
-		}
-		// if experimental mode is enabled, and devfile is provided.
-		// check if valid host is provided
-		if !pushtarget.IsPushTargetDocker() && len(o.host) <= 0 && (!o.isRouteSupported || o.wantIngress) {
-			return fmt.Errorf("host must be provided in order to create ingress")
+		if len(o.host) > 0 {
+			if o.urlType == envinfo.ROUTE {
+				errorList = append(errorList, "host is not supported for URLs of Route Kind")
+			}
+			if err := validation.ValidateHost(o.host); err != nil {
+				errorList = append(errorList, err.Error())
+			}
+		} else if o.urlType == envinfo.INGRESS {
```

Comment:
We should add some integration tests regarding the host validation and the error collection with multiple validation issue.

Reply:
I've added such tests but they fail because they essentially only cover things that are hidden behind the experimental mode because the possibility of multiple errors more or less only happens in that case.